### DISCUSSION TITLE
Always retrieving row 1 data when showing data on Row Side Panel

### DISCUSSION THREAD

**Discussion by lohhb** [Budibase/budibase#14437] [Help]
I have a Table of reports.
When I click any row on the table, the pop up Row Side Panel will always show data from row 1.
I close the Row Side Panel.
I click another row in the table, the pop up Row Side Panel will always show data from row 1.

Refer below 1st screen capture, clicked on row 6, customer is KLM but the Row Side Panel shows row1 customer ABC and it's data,
Closed the Row Side Panel.
2nd screen capture, I clicked row 9 customer STU, the pop up Row Side Panel will still show data from row 1 customer ABC and it's data.
On the 3rd picture, for the New Table [On row click], I only have one Action which is Open Side Panel.
What other Action did I missed?
The data is being fetch wrongly, so anyone can advise?
I do not know how to do the coding.
![Row 6](https://github.com/user-attachments/assets/7d86b33f-acf6-46e5-8b65-78bd107fe476)
![Row 9](https://github.com/user-attachments/assets/02789fc4-d849-4dc2-8c7b-b53a211f2927)
![Untitled](https://github.com/user-attachments/assets/96349335-2447-4133-b58f-175747800ad1)

**Comment by mikesealey** [comment]
Hey @lohhb 

For "On row click" actions you need to use a "Set State" action to hold the `_id` of that row. We can then pass that `_id` to the form in the side panel to specify that we want to update this row in particular.

It looks like you're using a Form Block to update the row, so you'll need to use this field to fetch the row `_id` from state.
![image](https://github.com/user-attachments/assets/1e430a26-eee2-49a5-8b50-cf0599a55fa4)

I think the UI has changed a little since this was written, but [this guide on app state](https://docs.budibase.com/docs/app-state#app-state-in-blocks) is a great place to start.

Hopefully this gets you un-stuck. If you've still got more questions you might also find help in the [Budibase community discord server.](https://discord.com/invite/ZepTmGbtfF)

**Reply by lohhb** [reply]
I tried to use "Update State" (there is no "Set State" action, maybe it got renamed.
Still it doesn't work.
![Update state](https://github.com/user-attachments/assets/165d75d0-e173-42c2-8143-ed5c76ae4484)
